pkg/diskmaker/diskmaker.go

```diff
@@ -203,14 +235,14 @@ func (d *DiskMaker) symLinkDisks(diskConfig *DiskConfig) {
 				klog.V(3).Infof("symlinking to %s to %s", deviceNameLocation.diskNamePath, symLinkPath)
 				symLinkErr = os.Symlink(deviceNameLocation.diskNamePath, symLinkPath)
 			}
-
 			if symLinkErr != nil {
 				msg := fmt.Sprintf("error creating symlink %s: %v", symLinkPath, symLinkErr)
 				e := newEvent(ErrorFindingMatchingDisk, msg, deviceNameLocation.diskNamePath)
 				d.eventSync.report(e, d.localVolume)
 				klog.Errorf(msg)
 			}
-
+			// unlock before proceeding
+			unlockFunc()
```

Comment:
if any of those `continue` statements are encountered that that will cause the lock to be released at the end of the function.

Reply:
Added change to include unlockfunc before each of the exit points of the loop. Will be in next push
Will open bug to refactor based on entry in our triaging google sheet (as discussed)